server/src/pages/shotindex/controller.js

```diff
@@ -25,31 +25,45 @@ exports.launch = function(m) {
     }
     Object.assign(m, extractQueryParamValues(queryParamModelPropertyMap));
     model = m;
-    render();
-    if (firstRun && m.shots === null) {
-      // The actual shot data hasn't been loaded yet, so we'll immediately request it:
-      refreshModel();
-    }
-    firstRun = false;
-    return;
   }
+
   if (window.wantsauth) {
-    if (window.wantsauth.getAuthData() && !location.search.includes("reloaded")) {
-      location.search = "reloaded";
-      return;
-    }
-    const authTimeout = setTimeout(() => {
-      // eslint-disable-next-line no-global-assign, no-native-reassign
-      location = location.origin + "/#tour";
-    }, FIVE_SECONDS);
-    window.wantsauth.addAuthDataListener((data) => {
-      if (location.search.indexOf("reloaded") > -1) {
+    // Handle non-owner my shots page redirection
+    if (!m.hasDeviceId) {
+      if (window.wantsauth.getAuthData() && !location.search.includes("reloaded")) {
+        location.search = "reloaded";
         return;
       }
-      clearTimeout(authTimeout);
-      location.search = "reloaded";
-    });
+      const authTimeout = setTimeout(() => {
+        // eslint-disable-next-line no-global-assign, no-native-reassign
+        location = location.origin + "/#tour";
+      }, FIVE_SECONDS);
+      window.wantsauth.addAuthDataListener((data) => {
+        if (location.search.indexOf("reloaded") > -1) {
+          return;
+        }
+        clearTimeout(authTimeout);
+        location.search = "reloaded";
+      });
+      return;
+    }
+
+    if (window.wantsauth.getAuthData()) {
+      Object.assign(model, window.wantsauth.getAuthData());
+    } else {
+      window.wantsauth.addAuthDataListener((data) => {
+        Object.assign(model, data);
+        render();
+      });
+    }
```

Comment:
This if-else will execute on `refreshModel()`, which is not necessary.

Reply:
@chenba Updated shotindex to getAuthData only if both req.accountId (value stored in model.hasFxa) and model.accountId is null, ensuring this code is not exceuted on refreshModel